### PR TITLE
[RTF-1401] SqlFullTextLanguage field type changed to enum

### PR DESCRIPTION
**Pull request by maria-kizhaeva**
Complete the following tasks before merging in your PR:
- [x] If this is a user facing change, update [version.txt](https://github.com/relativitydev/relativity.testing.framework/blob/master/version.txt) and [CHANGELOG.md](https://github.com/relativitydev/relativity.testing.framework/blob/master/CHANGELOG.md)

### CODE REVIEW

File: source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			FieldInfo field = value.GetType().GetField(value.ToString());
+			DescriptionAttribute attribute = Attribute.GetCustomAttribute(field, typeof(DescriptionAttribute)) as DescriptionAttribute;
+			return attribute == null ? value.ToString() : attribute.Description;
+		}
+
+		public static SqlFullTextLanguage FromDescription(string langName)
+		{
+			if (!_sqlFullTextLanguageDescriptionMap.ContainsKey(langName))
+			{
+				throw new InvalidEnumArgumentException("Language name not found.");
```
Comment (maria-kizhaeva) [inline source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs:33]: I was not sure whether we should just return some default value or alarm an issue. As far as I understood the old API returns the lang as string (description value) and we might need to know if the value that's coming is not listed on our side

File: source/Relativity.Testing.Framework/Models/SqlFullTextLanguage.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+﻿using System.ComponentModel;
+
+namespace Relativity.Testing.Framework.Models
+{
+	public enum SqlFullTextLanguage
```
Comment (maria-kizhaeva) [inline source/Relativity.Testing.Framework/Models/SqlFullTextLanguage.cs:5]: Those are the list of values we can retrieve as eligible langs by Relativity.Workspaces/workspace/eligible-sql-full-text-languages

File: source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+using Relativity.Testing.Framework.Models;
+
+namespace Relativity.Testing.Framework.Tests.Extensions
+{
+	[TestFixture]
+	[TestOf(typeof(SqlFullTextLanguageExtensions))]
+	public class SqlFullTextLanguageExtensionsFixture
+	{
+		[Test]
+		public void ToDescription_happyPath()
```
Comment (jonatanjakoniuk) [inline source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs:14]: capital H

File: source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+		{
+			string fullLangDescription = SqlFullTextLanguage.ChineseSingapore.ToDescription();
+			fullLangDescription.Should().Be("Chinese (Singapore)");
+		}
+
+		[Test]
+		public void FromDescription_happyPath()
```
Comment (jonatanjakoniuk) [inline source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs:21]: capital H

File: source/Relativity.Testing.Framework/GlobalSuppressions.cs [hunk 5]
```diff
@@ -16,3 +18,5 @@
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1819:Properties should not return arrays", Justification = "<Pending>", Scope = "member", Target = "~P:Relativity.Testing.Framework.Models.KeywordSearch.Fields")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1819:Properties should not return arrays", Justification = "<Pending>", Scope = "member", Target = "~P:Relativity.Testing.Framework.Models.View.Fields")]
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1720:Identifier contains type name", Justification = "<Pending>", Scope = "type", Target = "~T:Relativity.Testing.Framework.Models.TabLinkType")]
+[assembly: SuppressMessage("Minor Code Smell", "S3963:\"static\" fields should be initialized inline", Justification = "<Pending>", Scope = "member", Target = "~M:Relativity.Testing.Framework.Extensions.SqlFullTextLanguageExtensions.#cctor")]
```
Comment (jonatanjakoniuk) [inline source/Relativity.Testing.Framework/GlobalSuppressions.cs:21]: follow the pattern- use full namespace

File: source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		static SqlFullTextLanguageExtensions()
+		{
+			_sqlFullTextLanguageDescriptionMap = new Dictionary<string, SqlFullTextLanguage>();
+			foreach (SqlFullTextLanguage lang in Enum.GetValues(typeof(SqlFullTextLanguage)))
+			{
+				_sqlFullTextLanguageDescriptionMap.Add(lang.ToDescription(), lang);
+			}
+		}
+
+		public static string ToDescription(this SqlFullTextLanguage value)
```
Comment (jonatanjakoniuk) [inline source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs:22]: name ToDescription is directly connected to DescriptionAttribiute (used on enum) and it looks weird here
I would use standard ToString and FromString

File: source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs (hunk 1 above)
Comment (jonatanjakoniuk) [reply source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs:33]: I'm not a huge fan of throwing Exceptions but here I don't have strong opinion.
For sure we need to know value of `langName` which caused trouble - for now we can add it to Exception message

Comment (maria-kizhaeva) [reply source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs:33]: I wasn't sure about adding it, but ok

File: source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs (hunk 6 above)
Comment (maria-kizhaeva) [reply source/Relativity.Testing.Framework/Extensions/SqlFullTextLanguageExtensions.cs:22]: But it does return a description indeed. I don't mind a ToString but I'd use something other than standard functions

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [7.0.0] - 2021-10-01
+
+### Changed
+
+- SqlFullTextLanguage property  of Workspace model changed to Enum object.
```
Comment (briantoms) [inline CHANGELOG.md:12]: Extra space between `property` and `of`

- [ ] Fix whitespace

**Review by briantoms — APPROVED** [review]

File: source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+using Relativity.Testing.Framework.Models;
+
+namespace Relativity.Testing.Framework.Tests.Extensions
+{
+	[TestFixture]
+	[TestOf(typeof(SqlFullTextLanguageExtensions))]
+	public class SqlFullTextLanguageExtensionsFixture
+	{
+		[Test]
+		public void ToDescription_HappyPath()
```
Comment (jonatanjakoniuk) [inline source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs:14]: ToFullName

Comment (jonatanjakoniuk) [reply source/Relativity.Testing.Framework.Tests/Extensions/SqlFullTextLanguageExtensionsFixture.cs:14]: - [ ] Change test method names

**Review by briantoms — APPROVED** [review]

**Review by paulletang — DISMISSED** [review]
Changes look fine.
Do we have any other breaking changes that we want to make soon that we should combine this with before going to master?

**Comment by briantoms** [comment]
This is going to break RTF.API if consuming the new version of RTF. We should probably have a PR ready to go with the RTF.API changes before merging this.

**Review by jonatanjakoniuk — APPROVED** [review]

**Comment by maria-kizhaeva** [comment]
> Changes look fine. Do we have any other breaking changes that we want to make soon that we should combine this with before going to master?



> This is going to break RTF.API if consuming the new version of RTF. We should probably have a PR ready to go with the RTF.API changes before merging this.

@paulletang @briantoms 

Nothing breaking, but I wanted to bump RTF version for RTF.API as part of [RTf-1268](https://jira.kcura.com/browse/RTF-1268) work. I've already tested it with local nuget package and it looks fine. Only required to fix code in few places

**Review by briantoms — APPROVED** [review]

**Review by paulletang — APPROVED** [review]

**Pull request merged**